src/main/groovy/com/devsoap/vaadinflow/util/PathUtils.groovy

```diff
@@ -0,0 +1,39 @@
+/*
+ * Copyright 2018 Devsoap Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.devsoap.vaadinflow.util
+
+import org.gradle.api.Project
+
+/**
+ * Utility methods for resolving paths
+ *
+ * @author John Ahlroos
+ * @since 1.0
+ */
+class PathUtils {
+
+    /**
+     * Get the root directory for a project where the submodule path is also taken into account
+     *
+     * @param project
+     *      the project to get the directory for
+     * @return
+     *      the root directory of the project
+     */
+    static File getSubmoduleSensitiveProjectRootDir(Project project) {
+        new File(project.rootDir, project.path.replace(':', '/'))
```

Comment:
why not `project.projectDir`?
https://docs.gradle.org/current/dsl/org.gradle.api.Project.html#org.gradle.api.Project:projectDir

It is possible to have projects with name `foo` but in `modules/coolOnes/fooModule` directory

Reply:
Thanks for the review!

According to the docs `projectDir` points to the directory where *build.gradle* is located. 

What happens for projects which have a single *build.gradle* in the root directory and all modules are defined there, will that property still point to the sub-module root instead of the root directory?

Reply:
You are right, `project.projectDir` seems to work in all cases. Changed it. :+1: